tests/curators/test_cat_curators.py

```diff
@@ -192,6 +192,21 @@ def test_df_curator(df, categoricals):
         ln.Schema.filter().delete()
 
 
+def test_pass_artifact(df):
```

Comment:
One could consider parametrizing all tests with passing artifact vs. not passing it but given the minimal layer of logic in every constructor it's presumably overkill.

Another elegant way of sharing the logic across all Curator classes could be a private method

```
def _assign_dataset_and_artifact(data: Any):
            if isinstance(sdata, Artifact):
            self._artifact = sdata
            self._dataset = sdata.load()
        else:
            self._artifact = None
            self._dataset = sdata
```

Then we'd not even need to test it outside of testing this mini function.

Reply:
Then, all curator child classes could simply call this method, @sunnyosun?

Reply:
sure